plugins/BEdita/Core/src/Model/Schema/JsonSchema.php

```diff
@@ -0,0 +1,188 @@
+ * Copyright 2017 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace BEdita\Core\Model\Schema;
+
+use Cake\Datasource\Exception\RecordNotFoundException;
+use Cake\Network\Exception\NotFoundException;
+use Cake\ORM\TableRegistry;
+use Cake\Utility\Inflector;
+
+/**
+ * JSON Schema generation utilities
+ *
+ * Provides static methods to generate JSON Schema for objects and resources
+ */
+class JsonSchema
+{
+    /**
+     * Valid resource names
+     *
+     * @var array
+     */
+    const VALID_RESOURCES = [
+        'applications',
+        'roles',
+        'streams',
+    ];
+
+    /**
+     * Generate JSON Schema draft-6 (https://tools.ietf.org/html/draft-wright-json-schema-01)
+     *
+     * @param string $typeName Type name of a resource or object
+     * @param string $url Url of this schema
+     * @return array Associative array representing schema
+     */
+    public static function generate($typeName, $url)
+    {
+        $isResource = in_array($typeName, static::VALID_RESOURCES);
+        $schema = [];
+        if ($isResource) {
+            $schema = static::resourceSchema($typeName);
+        } else {
+            try {
+                $objectType = TableRegistry::get('ObjectTypes')->get($typeName);
+                $schema = static::objectSchema($objectType);
+            } catch (RecordNotFoundException $e) {
+                throw new NotFoundException(__d('bedita', 'Type "{0}" not found', $typeName));
+            }
+        }
+
+        $baseSchema = [
+            'definitions' => json_decode("{}"),
+            '$id' => $url,
+            '$schema' => 'http://json-schema.org/draft-06/schema#',
+            'type' => 'object',
+        ];
+
+        return array_merge($baseSchema, $schema);
+    }
+
+    /**
+     * Build resource properties directly from a db table
+     *
+     * @param string $name Resource type name
+     * @return array JSON Schema array with `properties` and `required`
+     */
+    public static function resourceSchema($name)
+    {
+        $table = TableRegistry::get((string)Inflector::camelize($name));
+        $entity = $table->newEntity();
+        $schema = $table->getSchema();
+        $hiddenProperties = $entity->hiddenProperties();
+
+        $properties = [];
+        $required = [];
+        foreach ($schema->columns() as $column) {
+            if (in_array($column, (array)$table->getPrimaryKey()) || in_array($column, $hiddenProperties)) {
+                continue;
+            }
+
+            $metadata = $schema->getColumn($column);
+            $properties[$column] = static::convertColumn($column, $metadata, $entity->accessible($column));
```

Comment:
`$entity->accessible()` is deprecated. One should use `$entity->isAccessible()` instead.

Reply:
I did, but **scrutinizer** was complaining... don't know why
But since it's deprecated I'll restore `isAccessible()`